### PR TITLE
TFE based on TF2

### PR DESCRIPTION
**Pull request by zjn-code**
tfe based on tf2
- eager execution
- auto add dependency for stateful op
- define a graph by `tfe.function`
- nearly the same performance as tf1
- build tfe model from onnx model
- fix device bug

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,82 +60,82 @@ def provide_input():
...
 
-# define inputs
+# provide inputs
 w = tfe.define_private_variable(tf.ones(shape=(10,10)))
 x = provide_input()
 
-# define computation
+# eager execution
 y = tfe.matmul(x, w)
+res = y.reveal().to_natie()
```
Comment (jvmncs) [inline README.md:69]:
```suggestion
res = y.reveal().to_native()
```

File: README.md [hunk 2]
```diff
@@ -49,82 +60,82 @@ def provide_input():
...
+
+# build graph and run graph
+@tfe.function
+def matmul_func(x, w)
+    y = tfe.matmul(x, w)
+    return y.reveal().to_natie()
```
Comment (jvmncs) [inline README.md:75]:
```suggestion
    return y.reveal().to_native()
```

File: examples/README.md [hunk 3]
```diff
@@ -2,12 +2,17 @@
...
-- [federated-learning](./federated-learning/) secure aggregation for federated learning
-- [house-credit-default](./house-credit-default/) private predictions with a logistic regression model
-- [logistic](./logistic/) private training of logistic regression model using stochastic gradient descent
-- [securenn](./securenn/) private predictions using several neural networks from the SecureNN paper
+- [application](./application/) examples of using TF Encrypted in various scenarios
+  - [simple-average](./application/simple-average/) very basic example computing an average to get started
+  - [federated-learning](./application/federated-learning/) secure aggregation for federated learning
+  - [house-credit-default](./application/house-credit-default/) private predictions with a logistic regression model
+  - [joint-inference](./application/joint-inference/) joint inference with model owner and prediction client
+  - [joint-train](./application/joint-train/) joint train with horizontally or vertically splited dataset
```
Comment (jvmncs) [inline examples/README.md:10]:
```suggestion
  - [joint-train](./application/joint-train/) joint train with horizontally or vertically split dataset
```

**Review by zicofish — APPROVED** [review]
Excellent work! An important milestone to advance TFE alongside Tensorflow.

**Pull request merged**